src/raglite/_chatml_function_calling.py

```diff
@@ -474,28 +473,31 @@ def chatml_function_calling_with_streaming(
         "created": completion["created"],
         "model": completion["model"],
         "choices": [
-            {
-                "finish_reason": "tool_calls",
-                "index": 0,
-                "logprobs": completion["choices"][0]["logprobs"],
-                "message": {
-                    "role": "assistant",
-                    "content": None,
-                    "tool_calls": [
-                        {
-                            "id": "call_" + f"_{i}_" + tool_name + "_" + completion["id"],
-                            "type": "function",
-                            "function": {
-                                "name": tool_name,
-                                "arguments": completion["choices"][0]["text"],
-                            },
-                        }
-                        for i, (tool_name, completion) in enumerate(
-                            zip(completions_tool_name, completions, strict=True)
-                        )
-                    ],
+            cast(
+                llama_types.ChatCompletionResponseChoice,
+                {
+                    "finish_reason": "tool_calls",
+                    "index": 0,
+                    "logprobs": completion["choices"][0]["logprobs"],
+                    "message": {
+                        "role": "assistant",
+                        "content": None,
+                        "tool_calls": [
+                            {
+                                "id": "call_" + f"_{i}_" + tool_name + "_" + completion["id"],
+                                "type": "function",
+                                "function": {
+                                    "name": tool_name,
+                                    "arguments": completion["choices"][0]["text"],
+                                },
+                            }
+                            for i, (tool_name, completion) in enumerate(
+                                zip(completions_tool_name, completions, strict=True)
+                            )
+                        ],
+                    },
                 },
-            }
+            )
```

Comment:
Instead of a cast, it would be better to actually construct the object here. That way we benefit from proper type checking.